### PR TITLE
add scorecard pipeline

### PR DESCRIPTION
**Pull request by UlisesGascon**
Refs: https://github.com/nodejs/nodejs.org/issues/5659

### CODE REVIEW

**Review by ovflowd — CHANGES_REQUESTED** [review]
Why do we need to run this on the Website codebase? Shouldn't this be executed against nodejs/node?

**Comment by AugustinMauroy** [comment]
@ovflowd it's scorecard of this repo so we need an action to analyse the code and output the score 😄

**Comment by ovflowd** [comment]
Sorry, but we don't need a GitHub Action in this repository to have the scorecard analysed here.

This is a simple GitHub Action. It can be done anywhere else. And I rather have this on the Security WGs repository.

Also, @AugustinMauroy I'm asking @UlisesGascon's feedback here, not yours. (Please avoid answering on behalf of the PR author)

**Comment by ovflowd** [comment]
Note that Im happy to be proven wrong, and to understand why we might need this 🤔

**Comment by UlisesGascon** [comment]
> Why do we need to run this on the Website codebase? Shouldn't this be executed against nodejs/node?

Yep @AugustinMauroy is correct here. Let me explain in more in details: 

The `nodejs/nodejs.org` repository has also a scoring, actually all the repositories in the org have them. There are two ways on how the Scorecard data is updated:
1. CRON: basically the OpenSSF runs a massive CRON job and pull the most relevant repositories from Github and execute the analysis.
2. Voluntary notification: The projects will proactively build the SARIF and upload it to the Github Dashboard.

While the CRON way is great for exploratory and build a great idea of the ecosystem for the community and the OpenSSF, it is not a reliable for us due the way we are monitoring the changes and the evolution of all the repos at organization level.

> This is a simple GitHub Action. It can be done anywhere else. And I rather have this on the Security WGs repository.

Until now, we had included in each repository that we actively monitored (we are increasing the scope now):
- [nodejs/undici](https://github.com/nodejs/undici/blob/main/.github/workflows/scorecard.yml)
- [nodejs/node](https://github.com/nodejs/node/blob/main/.github/workflows/scorecard.yml)

And even for some Node dependencies as well:
- [ada-url/ada](https://github.com/ada-url/ada/blob/main/.github/workflows/scorecard.yml)

The pipeline itself is quite simple, with a single dependency that doesn't require maintenance (dependabot will manage the upgrades) and it runs as a cron job, so it won't affect the current workflow of the team (even if fails).

**Comment by ovflowd** [comment]
> Yep @AugustinMauroy is correct here. Let me explain in more in details:

Never said he isn't.

**Comment by ovflowd** [comment]
> While the CRON way is great for exploratory and build a great idea of the ecosystem for the community and the OpenSSF, it is not a reliable for us due the way we are monitoring the changes and the evolution of all the repos at organization level.

Gotcha. Ack that. That's good reasoning enough for me. If this is a limitation, and it is per design that individual repositories should run that, then that's fine by me.

File: .github/workflows/scorecards.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        with:
+          egress-policy: audit
+
+      - name: "Checkout code"
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+        with:
+          persist-credentials: false
+
+      - name: "Run analysis"
+        uses: ossf/scorecard-action@99c53751e09b9529366343771cc321ec74e9bd3d # v2.0.6
```
Comment (ovflowd) [inline .github/workflows/scorecards.yml:45]: Whitelisted the Action

**Comment by ovflowd** [comment]
FYI file has linting issues, @UlisesGascon

File: .github/workflows/scorecards.yml (hunk 1 above)
Comment (UlisesGascon) [reply .github/workflows/scorecards.yml:45]: Thanks!

**Comment by UlisesGascon** [comment]
linting issues resolved in fb9829f

**Review by ovflowd — APPROVED** [review]

**Pull request merged**